### PR TITLE
LCAM-617| added dtos and calc_hardship method

### PR DESCRIPTION
**Pull request by AafAnsari**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-617)

Added `calculateHardship` method, dtos and unit tests

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/dto/HardshipReviewCalculationDTO.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.justice.laa.crime.hardship.dto;
+
+import lombok.Builder;
+import lombok.Data;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Data
+@Builder
```
Comment (loganathanMOJ) [inline crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/dto/HardshipReviewCalculationDTO.java:10]: Please add @AllArgsConstructor
@NoArgsConstructor

File: crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/service/HardshipService.java [hunk 2]
```diff
@@ -34,4 +45,32 @@ public ApiCalculateHardshipByDetailResponse calculateHardshipForDetail(ApiCalcul
...
+        if (!isEmpty(hardshipReviewCalculationDTO.getHardshipReviewCalculationDetails())) {
+            for (HardshipReviewCalculationDetail hRDetailDTO : hardshipReviewCalculationDTO.getHardshipReviewCalculationDetails()) {
+                if (Arrays.asList(INCOME, SOL_COSTS, EXPENDITURE).contains(hRDetailDTO.getDetailType())
+                        && BigDecimal.ZERO.compareTo(hRDetailDTO.getAmount()) != 0 && "Y".equals(hRDetailDTO.getAccepted())) {
+                    hardshipSummary = hardshipSummary.add(hRDetailDTO.getAmount()
+                            .multiply(BigDecimal.valueOf(hRDetailDTO.getFrequency().getAnnualWeighting())));
+                }
+            }
+        }
+        final BigDecimal disposableIncome = hardshipReviewCalculationDTO.getDisposableIncome();
```
Comment (loganathanMOJ) [inline crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/service/HardshipService.java:61]: Assign to Zero if hardshipReviewCalculationDTO.getDisposableIncome() is null.

Comment (AafAnsari) [reply crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/service/HardshipService.java:61]: done

File: crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/dto/HardshipReviewCalculationDTO.java (hunk 1 above)
Comment (AafAnsari) [reply crime-hardship/src/main/java/uk/gov/justice/laa/crime/hardship/dto/HardshipReviewCalculationDTO.java:12]: done

**Review by loganathanMOJ — APPROVED** [review]

**Pull request merged**